API: Extend IImage3dFileLoader interface with  IsHighPerfMode method

forderud opened the pull request:
Proposal for enabling run-time query if the loader is operating in high-perf (GPU) vs. lower-perf (CPU) mode. (issue https://github.com/MedicalUltrasound/Image3dAPI/issues/92).

SteveKauffman commented:
In the case that a particular loader is always CPU-based, what is the expected value of IsHighPerfMode?  I'd think that it would be 'true', as the expected usage of this API method is to communicate to the user, "we're operating at a lower performance level, but a higher performance level is possible."

Can the comments be updated accordingly?

forderud commented:
Thanks for your feedback @SteveKauffman. I've now rebased the PR, and modified DummyLoader to always return "true" in accordance to the doc.

This is an API change, so let's await merging until our call next week, so that we're sure that all vendors are ok with it.

forderud commented:
Decided to defer this API proposal for the time being. We can always reopen i later if/when needed.

SteveKauffman commented:
For decision documentation: deferring this API proposal was was decided by the joint technical team in meeting of 2018-09-19.  The reason for deferring was that:

- the lack of this API method is not blocking our next steps forward toward demo of other-vendor loaders on each vendor's workstation

- concern about how this would be used - if another company's loader is reporting that we're in low-performance mode, your service organization may not want to deal with support for resolving it
